jupyter_sphinx/execute.py

```diff
@@ -394,11 +399,24 @@ def apply(self):
                 source = node.children[0]
                 source.attributes['language'] = lexer
 
-            # Add line numbers
+            # Add line numbers to code cells if jupyter_sphinx_linenos or
+            # jupyter_sphinx_continue_linenos are set in the configuration,
+            # or the linenos directive is set.
+            # Reset linenumber if linenos directive is set.
+            # Update current line numbers from cell if jupyter_sphinx_continue_linenos
+            # is set.
+            linenostart = 1
             for node in nodes:
+                source = node.children[0]
+                if linenos_config or continue_linenos:
+                    source["linenos"] = True
                 if node["linenos"]:
-                    source = node.children[0]
+                    linenostart = 1
```

Comment:
I think this would be a confusing interaction: it's best if the option `continue_linenos` works the same regardless of whether line numbers are turned on globally or not.

Reply:
Not sure if I am following you. This allows restart from one if :linenos: directive is added to a cell and `jupyter_sphinx_continue_linenos` config option is set. It does not matter if `linenos` is turned on as a directive or by the `jupyter_sphinx_linenos` global option.

Reply:
I'm saying that a document with every input containing `:linenos:` should always render exactly the same as if the config option `jupyter_sphinx_linenos` was set.

If I'm not mistaken, right now if the user sets `jupyter_sphinx_continue_linenos`, and all cells contain `:linenos:` option, the line numbers would be reset in each cell, resulting in a different output.

Reply:
Ok. That makes sense. I took that out. I think it would be nice to be able to reset the numbering, however. Would a :linenos-reset: directive be okay?  I don't there if there is a way to do "magic" that would accomplish the same thing. It could be ":_linenos-reset".  :)

Reply:
I feel that any extra setting would be too niche. Would it be possible to achieve a similar result by resetting the count on any hidden input?

Reply:
Okay, let's hold off on that. This can go merge as is without it.

Reply:
For what it's worth, I wanted exactly this reset feature as soon as I started using the `continue_linenos` `conf.py` setting. I'm disappointed it was taken out.

Reply:
Another way of doing a restart of the numbering restart would be to add support for the Sphinx code-block directive`` lineno-start``, which could be set to 1 if continuation is on. I think that would be generally useful, e.g. for referencing earlier code, and intuitive to use. https://www.sphinx-doc.org/en/1.5/markup/code.html#line-numbers